YKI(Frontend): OPHKIOS-184 list exams

saku-koodari opened the pull request:
## Yhteenveto

_Toteutuksen yhteenveto..._

## Lisätiedot

Listattu testit: ilmoittautumiset / Jonossa / Menneet
<img width="872" height="760" alt="image" src="https://github.com/user-attachments/assets/8d7be411-41d6-4e2b-8366-dead30b22347" />


## Huomautukset

Tältä sivulta puuttuu vielä:
 * Muokkaa yhteystietoja
 * Sorttaus
 * Päivämäärät Dayjs formaatissa:
   * en tehnyt tässä, koska nyt API:n tietomallia käytetään reduxissa sellaisenaan, ja se pitäisi erotella eri luokkaan, joka serialisoidaan
   * Toiminnot - kolumni. Mun mielestä tätä ei ole vielä mielekästä tehdä, kun ei ole bäkkäriä. Parempi tehdä bäkkäri-toteutuksen yhteydessä tai sen jälkeen. Nyt edes tästä on käytettävissä ns. `read-only` - versio
   * Tutkintopäivämäärä on ilmeisesti linkki. Mihin sen linkin pitäisi johtaa?

## Check-lista

- [ ] yarn.lock päivitetty
- [ ] Käännösten Excel-tiedosto päivitetty
- [ ] UI-muutoksia testattu eri selaimilla ja mobiilinäkymässä

saku-koodari committed:
YKI(Frontend): move CustomerInformation to an own file

saku-koodari committed:
wip: YKI(Frontend): Show dates of each exam list

saku-koodari committed:
YKI(Frontend): Use correct import

saku-koodari committed:
add language resources

saku-koodari committed:
show header texts

saku-koodari committed:
add more columns

new columns:
 - examLocation
 - registrationState
 - registrationDate
 - queueSpotOffered

fix title of examLocation

saku-koodari committed:
put schoolname and municipality into separate rows

saku-koodari committed:
fix language and languageLevel

saku-koodari committed:
show paidAt value if the status is completed

saku-koodari committed:
add stylings to registrationStates

saku-koodari committed:
add examState

saku-koodari committed:
fixes to header and spacing

saku-koodari committed:
add tutkinto

saku-koodari committed:
show red text if offerStatus is not accepted

saku-koodari committed:
wip: cypress tests

saku-koodari committed:
fix rendering when there are no exams

saku-koodari committed:
finish empty data test

saku-koodari committed:
test headers

saku-koodari committed:
test exams data

saku-koodari committed:
update mock data

kimmotaskinen-knowit commented on frontend/packages/yki/public/i18n/en-GB/public.json:
```diff
@@ -182,6 +182,57 @@
+          },
+          "listing": {
+            "headers": {
+              "registratedExams": "Ilmoittautumiset",
```
projektin domain language on ilmoittautumisille ollut ihan vaan “registrations”.

Tässä näkymässä pari konseptia yhdessä niin vähän sekoittaa kun linkki johtanee sitten kuitenkin tutkintotilaisuuden näkymään mutta jos kälissä halutaan pitää toi otsikko “Ilmoittautumiset” niin sitten parempi toi “registrations” niin linjassa aikasempien kanssa

kimmotaskinen-knowit commented on frontend/packages/yki/public/i18n/en-GB/public.json:
```diff
@@ -182,6 +182,57 @@
+              "pastExams": "edeltävät 365pv"
+            },
+            "noRowsTexts": {
+              "registratedExams": "Ei tulevia ilmoittautumisia",
```
Nää ois ehkä sisitimpe jos vaan “noRegistrations..” etc (sama korjaus tänne ku tonne ylös kun puhutaan ilmoittautumisista

saku-koodari replied:
laitoin tän vaan muotoon `registeratations`, koska periaatteessa nuo jsonit vois uudelleen organisoida:

```json
{
  "registrations": {
    "header": "Ilmoittautumiset",
    "noRowsText": "Ei tulevia ilmoittautumisia",
    // muita registrations - kohtaisia kenttiä...
  }
}
```

Toi sama patterni pätee myös `Jonossa` ja `Menneet` - tutkinnoille.

kimmotaskinen-knowit commented on frontend/packages/yki/src/components/clerkCustomer/CustomerExamListings.tsx:
```diff
@@ -0,0 +1,295 @@
+  subHeader: string;
+  noRowsText: string;
+}) => {
+  const hasNoRows = !rows || rows.length == 0;
```
Nää voisi pitää strict equaleina. Näköjään hieman vuotanut kun 753 matchia ‘===‘ ja 50 matchia ‘==‘ projektissa tällä hetkellä. Vois tehdä pienen cleanup PR:n jossain välissä

saku-koodari replied:
hups. Nyt tuli lipsahdus

kimmotaskinen-knowit commented on frontend/packages/yki/src/components/clerkCustomer/CustomerExamListings.tsx:
```diff
@@ -0,0 +1,295 @@
+  return (
+    <div>
+      <div className="columns flex-start">
+        <H3 style={{ marginRight: '0.25em' }}>{header}</H3>
```
ton margin stylen voinee korvata ja lisätä vaan välilyönnin {‘ ‘}

kimmotaskinen-knowit replied:
Design kans näyttäs et tossa on vaan toi koko teksti tolla samalla typefacella mut se on kyl liian bold varmasti. Ei tarvi täs säätää mut kysäsen Artolta onks tossa figmassa jotain hidden tyylejä ku noi o noin erilaisia

saku-koodari replied:
välilyönti ei toimi, nimenomaan tuossa kohtaa. Lintteri linttaa

```diff
-<span>{' '}</span>
+<span> </span>
```

Mutta `&nbsp;` lisää riittävästi väliä, niin voinen käyttää sitä. Mulla on kaks muuta vastaavaa muuta ja yhdessä niistä lintteri ei poista tuota väliä. Käytän mielummin tuota `&nbsp;` kaikissa kohdissa.

saku-koodari committed:
YKI(Frontend): pr fixes:
 - rename registratedExams to registrations
 - use false check in ExamsListing no rows check
 - use &nbsp; instead of margin-right to give space between elements
 - use indicator colors instead of hardcoded
 - lint fix

kimmotaskinen-knowit commented on frontend/packages/yki/src/components/clerkCustomer/CustomerExamListings.tsx:
```diff
@@ -0,0 +1,291 @@
+    title: t('columns.date'),
+    render: ({ examinationDate }) => (
+      <div className="rows gapped-xs">
+        {/* TODO: Convert registrationDate to DayJs */}
```
Heitä tolle `DateUtils.formatOptionalDate` funkkarille vaan toi string value niin voit poistaa todo

saku-koodari replied:
Yritin tuossa kuvauksessa sanoa, että sulla tuo toimii, koska noi FreeRegistration - date - kentät on tyyppiä `DayJs`.

Siellä on tehty niin, että sagassa [vastaanotetaan](https://github.com/Opetushallitus/kieli-ja-kaantajatutkinnot/blob/8b9e270dbfce9e7f410d48c33b95accbbb4dd6ce/frontend/packages/yki/src/redux/sagas/clerkFreeRegistration.ts#L45) `ClerkFreeRegistrationDetailsResponse`, joka serialisoidaan, ja se konvertoi string -> DayJs. Tuossa Customerin - sagassa [ei oo ollenkaan serialoisointia](https://github.com/Opetushallitus/kieli-ja-kaantajatutkinnot/blob/8b9e270dbfce9e7f410d48c33b95accbbb4dd6ce/frontend/packages/yki/src/redux/sagas/clerkCustomerDetails.ts#L16)(, koska oon ollut laiska enkä ole halunnut vielä kirjoittaa erikseen Response ja frontin DTO(mikälie) - luokkaa)

saku-koodari replied:
Eli siis esim reduxin käyttämä `ClearFreeRegistrationDetails` [sisältää](https://github.com/Opetushallitus/kieli-ja-kaantajatutkinnot/blob/8b9e270dbfce9e7f410d48c33b95accbbb4dd6ce/frontend/packages/yki/src/interfaces/clerkFreeRegistration.ts#L104-L136) Dayjs - kenttiä. Ja axioksen käyttämästä responsesta erikseen omitoitu Dayjs - kentät ja ne korvattu string - kentillä.

Customers - tyypillä on _vastaavaan_ vaan [yks tyyppi,](https://github.com/Opetushallitus/kieli-ja-kaantajatutkinnot/blob/8b9e270dbfce9e7f410d48c33b95accbbb4dd6ce/frontend/packages/yki/src/interfaces/clerkCustomer.ts#L33-L66) jota käytetään sekä axioksen vastauksissa, että reduxissa.

kimmotaskinen-knowit replied:
Jepu, tsekataan vaan huddlessa toi ku se on tosi nopee heittää siihen ja sit helppo extendaa jos/kun tulee muuta serialisoitavaa

kimmotaskinen-knowit commented on frontend/packages/yki/src/components/clerkCustomer/CustomerExamListings.tsx:
```diff
@@ -0,0 +1,291 @@
+    Record<RegistrationStates, JSX.Element>
+  > = {
+    [RegistrationStates.Completed]: (
+      <CheckCircleIcon sx={{ color: 'green', fontSize: '1.2em' }} />
```
Näille vois antaa propsit `fontSize=“large”`ja color=“success” <= kato about lähellä oleva noista mitä ide tarjoaa: error, warning etc. Päivitetään sit themeen tarvittaessa noi kaikille kerralla, toi base theme ei oo ihan sama mitä designis mut näyttäs et kuitenkin semantic success se haluis siihen väriksi

kimmotaskinen-knowit commented on frontend/packages/yki/src/components/clerkCustomer/CustomerExamListings.tsx:
```diff
@@ -0,0 +1,282 @@
+    render: ({ registrationStatus }) => {
+      return (
+        <div className="rows gapped-xs">
+          <Text style={{ display: 'flex' }}>
```
Nää saanee parhaiten kuntoon jos heität tolla ListTable:lle className samaan tyyliin ku:
`className="clerk-free-registration-listing__table”` ja antaa sitä kautta td:lle `vertical-align: top;`

Tolta teksiltä vois ottaa ton style määrityksen veke ja siirrä se et wrappaa vaan noi tekstit tos alempana.

Lopuksi sitten jos tohon riville 150 heität “columns gapped-xs” ni pitäs olla toi paketti nätisti kasassa, ikonit ja tekstit keskitettynä ja top aligned

saku-koodari committed:
YKI(Frontend): Separate Response and Redux DTO in order to localize dates

saku-koodari committed:
YKI(Frontend): fix table cell alignments

kimmotaskinen-knowit committed:
YKI(Frontend): response example

saku-koodari committed:
YKI(Frontend): Add align table cell vertically from top

saku-koodari committed:
YKI(Frontend): Remove unused response

kimmotaskinen-knowit committed:
YKI(Frontend): Updated response types

kimmotaskinen-knowit reviewed (approved):
Je, pistä vaan sisään! Huomasin et toi Icon ja Text placement ei itseasiassa ollut enää tuolla maksuttomuudenkaan puolella ihan linjassa mut säädetään tota jossain vaiheessa
